components/icon/icon.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <component
+    :is="icon"
+    data-qa="dt-icon"
+    :class="iconSize"
+  />
+</template>
+
+<script>
+import { ICON_SIZE_MODIFIERS } from './icon_constants';
+import { kebabCaseToPascalCase } from '@/common/utils';
+
+/**
+ * The Icon component provides a set of glyphs and sizes to provide context your application.
+ * @see https://dialpad.design/components/icon.html
+ */
+export default {
+  name: 'DtIcon',
+
+  props: {
+    /**
+     * The size of the icon.
+     * @values 100, 200, 300, 400, 500, 600, 700, 800
+     */
+    size: {
+      type: String,
+      default: '500',
+      validator: (s) => Object.keys(ICON_SIZE_MODIFIERS).includes(s),
+    },
+
+    /**
+     * The icon name in kebab-case
+     */
+    name: {
+      type: String,
+      required: true,
+    },
+  },
+
+  computed: {
+    iconSize () {
+      return ICON_SIZE_MODIFIERS[this.size];
+    },
+
+    iconName () {
+      return kebabCaseToPascalCase(this.name);
+    },
+
+    icon () {
+      return require(`@dialpad/dialtone/lib/dist/vue/v7/${this.iconName}`).default;
+    },
```

Comment:
@braddialpad looks like with this I'm importing all the Icons from Dialtone and that increased bundle size.

Reply:
Changed require for imports, that should improve the bundle size

Reply:
Ugh looks like it might be even worse now for some reason.. 182% increase. I don't really have a good solution for this at the moment we'll have to look again later

Reply:
I've been looking at this all day... 

WebPack is trying to bundle the icons coming from `dialtone` when I `require` and chunks every Icon into its single file when `importing`.

Tried a lot of solutions and none of them are working correctly:

- Adding `@dialpad/dialtone/lib/dist/vue/v7` to `config.externals`, it does nothing different.
- Tell webpack to ignore those imported modules with `import(/* webpackIgnore: true */ ...);` I get a "Failed to resolve module specifier" error.

Maybe exporting the icons in a different way from `dialtone` can work?

Reply:
Yeah I have dealt with these tree shaking issues before and they are never easy to figure out.